src/Microsoft.DotNet.XHarness.Apple/AppRunnerBase.cs

```diff
@@ -45,5 +53,32 @@ protected async Task<string> FindDevice(TestTargetOs target)
 
             return companionDevice?.Name ?? device.Name;
         }
+
+        protected Task<ProcessExecutionResult> RunMacCatalystApp(
+            AppBundleInformation appInfo,
+            TimeSpan timeout,
+            IEnumerable<string> appArguments,
+            Dictionary<string, object> environmentVariables,
+            CancellationToken cancellationToken)
+        {
+            var binaryPath = Path.Combine(appInfo.AppPath, "Contents", "MacOS", appInfo.AppName);
+            var arguments = new List<string>();
+
+            if (!File.Exists(binaryPath))
+            {
+                _mainLog.WriteLine($"Failed to find an executable binary at {binaryPath}. Trying to run app using `open -W`");
```

Comment:
We can parse the Info.plist file in the appbundle and check `CFBundleExecutable` for a reliable way to get the binary name. I'm fine if we do this in a follow-up though :)